src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```
@@ -9694,60 +10071,621 @@ class Program
+        /// Ensure that the synthesized switch hash method can handle a null
+        /// value if null is supported as a constant pattern in the future.
+        /// </summary>
+        [Fact]
+        public void PatternMatchSpanChar_SwitchHashWithNull()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static int F1(ReadOnlySpan<char> span)
+    {
+        return span switch
+        {
+            (string)null => 0,
+            ""1"" => 1,
+            ""2"" => 2,
+            ""3"" => 3,
+            ""4"" => 4,
+            ""5"" => 5,
+            ""6"" => 6,
+            _ => 7,
+        };
+    }
+    static int F2(Span<char> span)
+    {
+        return span switch
+        {
+            ""1"" => 1,
+            ""2"" => 2,
+            ""3"" => 3,
+            ""4"" => 4,
+            ""5"" => 5,
+            ""6"" => 6,
+            default(string) => 7,
+            _ => 0,
+        };
+    }
+}";
+            var comp = CreateCompilationWithSpanAndMemoryExtensions(source, parseOptions: TestOptions.RegularNext);
+            comp.VerifyEmitDiagnostics(
+                // (8,13): error CS9013: A string 'null' constant is not supported as a pattern for 'ReadOnlySpan<char>'. Use an empty string instead.
+                //             (string)null => 0,
+                Diagnostic(ErrorCode.ERR_PatternSpanCharCannotBeStringNull, "(string)null").WithArguments("System.ReadOnlySpan<char>").WithLocation(8, 13),
+                // (28,13): error CS9013: A string 'null' constant is not supported as a pattern for 'Span<char>'. Use an empty string instead.
+                //             default(string) => 7,
+                Diagnostic(ErrorCode.ERR_PatternSpanCharCannotBeStringNull, "default(string)").WithArguments("System.Span<char>").WithLocation(28, 13));
+        }
+
+        /// <summary>
+        /// DecisionDagRewriter.EnsureStringHashFunction() does not generate
+        /// a hash function if the span indexer is missing.
```

Should `DecisionDagRewriter.EnsureStringHashFunction()` report an error if the indexer is missing rather than silently falling back to the non-hash implementation?